### PR TITLE
Added general erc-20 token transfers schema.

### PR DESCRIPTION
**Pull request by jamelvin**

### CODE REVIEW

File: schemas/general/schema.json [hunk 1]
```diff
@@ -98,6 +98,47 @@
...
       }
     },
+    {
+      "label": "Token ERC-20 Transfers",
+      "tableName": "general_token_erc20_transfers",
+      "aggregation": "transaction",
+      "uniqueKey": ["user_address", "token_address", "block_date"],
+      "description": "User level token balance snapshots.",
+      "properties": {
+        "block_time": {
```
Comment (melotik) [inline schemas/general/schema.json]: use `timestamp`

File: schemas/general/schema.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "schema": "General",
   "description": "Table definitions for the generic schema. These tables can be used for any protocol.",
-  "protocolCategory": ["Bond", "Gaming", "Oracle", "Payment", "Quest", "SocialFi", "Stablecoin", "Wallet"],
+  "protocolCategory": ["Bond", "Gaming", "LRT", "Oracle", "Payment", "Quest", "SocialFi", "Stablecoin", "Wallet", "Other"],
```
Comment (melotik) [inline schemas/general/schema.json]: Can undo this change

File: schemas/erc20/schema.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+  "schema": "ERC-20",
+  "description": "Table definitions for the ERC-20 schema. These tables can be used for any protocol.",
+  "protocolCategory": ["Bond", "Gaming", "LRT", "Oracle", "Payment", "Quest", "SocialFi", "Stablecoin", "Wallet", "Other"],
```
Comment (melotik) [inline schemas/erc20/schema.json]:
```suggestion
  "protocolCategory": ["ERC20"],
```

File: schemas/erc20/schema.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  "version": "1.0.0-alpha",
+  "tables": [
+    {
+      "label": "Token ERC-20 Transfers",
+      "tableName": "general_token_erc20_transfers",
+      "aggregation": "transaction",
+      "uniqueKey": ["user_address", "token_address", "block_date"],
+      "description": "User level token balance snapshots.",
+      "properties": {
```
Comment (melotik) [inline schemas/erc20/schema.json:13]: Let's add `block_date`, `transaction_hash`, and `log_index`

Need the last 2 for a uniqueKey, and the first for partitions

File: schemas/erc20/schema.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  "schema": "ERC-20",
+  "description": "Table definitions for the ERC-20 schema. These tables can be used for any protocol.",
+  "protocolCategory": ["Bond", "Gaming", "LRT", "Oracle", "Payment", "Quest", "SocialFi", "Stablecoin", "Wallet", "Other"],
+  "version": "1.0.0-alpha",
+  "tables": [
+    {
+      "label": "Token ERC-20 Transfers",
+      "tableName": "general_token_erc20_transfers",
+      "aggregation": "transaction",
+      "uniqueKey": ["user_address", "token_address", "block_date"],
```
Comment (melotik) [inline schemas/erc20/schema.json]:
```suggestion
      "uniqueKey": ["transaction_hash", "log_index"],
```

**Review by melotik — CHANGES_REQUESTED** [review]
Thanks for posting this PR!

The other thing we can think about here is having an tokens table, like this: https://github.com/delta-hq/schemas/blob/344fc6c38bc8001c031bfe8a7463591a758dac6c/schemas/dex-aggregator/schema.json#L14

File: schemas/erc20/schema.json (hunk 4 above)
Comment (jamelvin) [reply schemas/erc20/schema.json:13]: `timestamp` = `block_date` here.  Does that work okay or do I need to change the name?

Comment (melotik) [reply schemas/erc20/schema.json:13]: Actually, not needed, this setup works

File: schemas/erc20/schema.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  "description": "Table definitions for the ERC-20 schema. These tables can be used for any protocol.",
+  "protocolCategory": ["ERC20"],
+  "version": "1.0.0-alpha",
+  "tables": [
+    {
+      "label": "Token ERC-20 Transfers",
+      "tableName": "general_token_erc20_transfers",
+      "aggregation": "transaction",
+      "uniqueKey": ["transaction_hash", "log_index"],
+      "description": "User level token balance snapshots.",
```
Comment (melotik) [inline schemas/erc20/schema.json]: nit: this should be saying something about erc20 trasfer events

**Review by melotik — APPROVED** [review]
lgtm, thanks! feel free to merge after fixing the last comment

File: schemas/erc20/schema.json (hunk 6 above)
Comment (jamelvin) [reply schemas/erc20/schema.json]: sorry, missed that part
